### PR TITLE
FIX: `odr.Model` error with default `meta` value

### LINKED ISSUE #16500: odr.Model default meta value fails with __getattr__
The default value ``None`` for meta is not iterable. The method ``__getattr__`` will fail.
One could change the line
https://github.com/scipy/scipy/blob/4cf21e753cf937d1c6c2d2a0e372fbc1dbbeea81/scipy/odr/_odrpack.py#L521

with 
```python
self.meta = [] if meta is None else meta
```

### PR DESCRIPTION
**Pull request by eshirvana**
#### Reference issue

Closes gh-16500

### CODE REVIEW

**Review by tupui — COMMENTED** [review]
Thank you for the PR @eshirvana. LGTM, it might benefit from adding or modifying an existing test. Could you have a look?

**Comment by eshirvana** [comment]
@tupui test added , also Instead if statement , I simply set the default value for meta as I see this is more pythonian and that's actually how is done other classes

File: scipy/odr/tests/test_odr.py
Comment (tupui) [inline scipy/odr/tests/test_odr.py:532]: Instead I would test going through the iterator as in the issue.

**Review by tupui — CHANGES_REQUESTED** [review]
> @tupui test added , also Instead of if statement, I simply passed the default value for meta as I see this is more pythonian and that's actually how it is done in other classes in odr pack

No it's not Pythonic and actually a bad practice to use empty dict or list as default arguments (see the [warning](https://docs.python.org/3/tutorial/controlflow.html#default-argument-values) in the Python doc). If that's the case in other places, it should be changed to None as default.

File: scipy/odr/tests/test_odr.py
Comment (eshirvana) [reply scipy/odr/tests/test_odr.py:532]: @tupui  I'm not sure how to do the test using iterator, can you show me an example please?

Comment (rkern) [reply scipy/odr/tests/test_odr.py:532]: The `.meta` attribute being a `dict` is part of the public interface, not just an internal implementation detail. Testing that it is equal to `{}` in this case is a perfectly good thing to test. But it also would be a good idea to test that the other public functionality on `Model` that internally accesses `.meta` works, too. So test that `__getattr__()` behaves appropriately and `set_meta()`.

Comment (eshirvana) [reply scipy/odr/tests/test_odr.py]: @rkern  got it, thanks , updated the test  as well as the code.

**Comment by tupui** [comment]
@eshirvana I am sorry, but I am only realizing now that this PR should not have been opened. @boatwrong expressed interest before (and assume he was opening this PR) and we don't want to create a competitive environment.

As we see right now, it's creating a lot of extra work for maintainers and not creating a nice welcoming atmosphere for newcomers.

I will wait for a confirmation on the other PR before closing this one.

**Comment by eshirvana** [comment]
@tupui that's totally fine with me , no worries . I just appreciate if you review and let me know if the code is prod ready ( I know eventually you might go with other PR ) . like to know your feedback as its a practice for me to hopefully contribute more in future.
thanks

**Comment by tupui** [comment]
Thank you for your understanding. Once we decide on the PR, you can see the end result. As I tried to explain, we don't want to duplicate reviewing efforts.

**Review by rkern — APPROVED** [review]
LGTM once the CI passes. Thanks, @eshirvana and @boatwrong for cleaning up my errors.

**Review by tupui — APPROVED** [review]

**Comment by tylerjereddy** [comment]
It is labeled as a defect and cleanly cherry-picks onto maintenance so adding a tentative backport label.

**Pull request merged**
